cmd/lakectl/cmd/local_init.go

```diff
@@ -20,6 +22,25 @@ const (
 )
 
 func localInit(ctx context.Context, dir string, remote *uri.URI, force, updateIgnore bool) (string, error) {
+	client := getClient()
+	if len(remote.GetPath()) > 0 { // Verify path is not an existing object
+		stat, err := client.StatObjectWithResponse(ctx, remote.Repository, remote.Ref, &api.StatObjectParams{
+			Path: *remote.Path,
+		})
+		switch {
+		case err != nil:
+			DieErr(err)
+		case stat.JSON200 != nil:
+			DieFmt("lakeFS path %s is an existing object and cannot be used as a reference source", remote.String())
+		case stat.JSON404 == nil:
+			dieOnResponseError(stat, err)
+		}
```

Comment:
Can we use `DieOnErrorOrUnexpectedStatusCode`?

Reply:
Unfortunately no - since here we are expecting a status code which is "not" OK (not 2xx) we will fail. 
Unless we decide to change this logic and allow expecting also other status codes